src/Compilers/CSharp/Portable/Symbols/Source/CustomModifierUtils.cs

```diff
@@ -102,7 +102,8 @@ internal static TypeSymbol CopyTypeCustomModifiers(TypeSymbol sourceType, TypeSy
 
             Debug.Assert(resultType.Equals(sourceType, TypeCompareKind.IgnoreDynamicAndTupleNames | TypeCompareKind.IgnoreNullableModifiersForReferenceTypes)); // Same custom modifiers as source type.
 
-            Debug.Assert(resultType.Equals(destinationType, TypeCompareKind.IgnoreCustomModifiersAndArraySizesAndLowerBounds)); // Same object/dynamic, nullability and tuple names as destination type.
+            // Same object/dynamic, nullability and tuple names as destination type.
+            Debug.Assert(resultType.Equals(destinationType, TypeCompareKind.IgnoreCustomModifiersAndArraySizesAndLowerBounds | TypeCompareKind.IgnoreNativeIntegers));
```

Comment:
> | TypeCompareKind.IgnoreNativeIntegers [](start = 124, length = 39)

This doesn't feel right. Native int types specified in source should be preserved.

Reply:
@cston can you comment? Regardless I don't want to make a change around native int encoding in this PR, so I'm either going to skip the tests that hit this assert or keep this as is.

---
In reply to: [426089925](https://github.com/dotnet/roslyn/pull/44257#discussion_r426089925) [](ancestors = 426089925)

Reply:
It sounds like a bug. Please log an issue, thanks.

---
In reply to: [426823520](https://github.com/dotnet/roslyn/pull/44257#discussion_r426823520) [](ancestors = 426823520,426089925)

Reply:
Filed [https://github.com/dotnet/roslyn/issues/44358](https://github.com/dotnet/roslyn/issues/44358).

---
In reply to: [426836636](https://github.com/dotnet/roslyn/pull/44257#discussion_r426836636) [](ancestors = 426836636,426823520,426089925)

Reply:
Removed the flag and skipped the relevant test.

---
In reply to: [426894790](https://github.com/dotnet/roslyn/pull/44257#discussion_r426894790) [](ancestors = 426894790,426836636,426823520,426089925)